packages/edit-site/src/components/post-list/index.js

```diff
@@ -372,15 +372,14 @@ export default function PostList( { postType } ) {
 			const { getEntityRecord, getPostType, canUser } =
 				select( coreStore );
 			const siteSettings = getEntityRecord( 'root', 'site' );
-			const postTypeObject = getPostType( postType );
 			return {
 				frontPageId: siteSettings?.page_on_front,
 				postsPageId: siteSettings?.page_for_posts,
 				labels: getPostType( postType )?.labels,
-				canCreateRecord: canUser(
-					'create',
-					postTypeObject?.rest_base || 'posts'
```

Comment:
Nice simplification 👍 

I wonder why the pre-existing indirection and the default `'posts`` was necessary 🤔

Reply:
I'm guessing to avoid passing `undefined` when `postTypeObject` hasn't been resolved yet, which would result in sending a request to the general namespace. However, I'm unsure why the `posts` fallback was selected.

I fixed a similar issue in https://github.com/WordPress/gutenberg/pull/63423#discussion_r1673836639.

Reply:
Back in the day, when we were writing up custom admin menus, for example, and we were displaying labels for custom post types, we were falling back to the default labels for post types - because when registering a custom post type, not all labels are guaranteed to be declared. Maybe this assumed that the `rest_base` could not be declared, too.

Reply:
The `rest_base` should default to the [post type slug](https://developer.wordpress.org/reference/functions/register_post_type/#:~:text=Default%20is%20%24post_type.).

Reply:
True. But then it can be overridden 🤷‍♂️

Reply:
You can do almost everything in WP 😅

Reply:
Yup. I'm just trying to understand why this was here. Maybe it was a temporary thing as we were working to make the posts list more flexible in the future? It's a recent addition (#62705), perhaps @ntsekouras could remember.

Reply:
Yeah, the default was to avoid the `undefined` value. Regarding why we need the resource, it's because we are using the same component for `posts and pages` and will probably be the base component of other post types too. Finally regarding the `default:posts`, didn't think it mattered much :).

Aside this, I'm catching up with @Mamaduka's work on these parts and is great! Thank you!